src/plugins/gui/scenes/FiatPluginWebView.tsx

```diff
@@ -35,10 +37,18 @@ export function FiatPluginWebViewComponent(props: Props): JSX.Element {
     if (onMessage != null) onMessage(data, injectJs)
   })
 
-  React.useEffect(() => () => {
-    // Cleanup code when scene unmounts
-    if (onClose != null) onClose()
-  })
+  // React.useEffect(() => () => {
+  //   // Cleanup code when scene unmounts
+  //   if (onClose != null) onClose()
+  // })
+
+  React.useEffect(() => {
+    navigation.addListener('beforeRemove', event => {
+      if (onClose != null) {
+        if (onClose() === false) event.preventDefault()
+      }
+    })
```

Comment:
The `navigation.addListener` returns a cleanup, function, which you should hand to `useEffect`. So maybe just remove the `{}` around the `navigation.addListener` call, so the return value flows out.